app/controllers/concerns/course/statistics/submissions_concern.rb

```diff
@@ -89,8 +89,8 @@ def populate_hash_including_answers(student_hash, submissions)
       next unless submitter_course_user&.student?
 
       answers = answers_hash[submission.id]
-      end_at = @personal_end_at_hash[[@assessment.id, submitter_course_user.id]] ||
-               @reference_times_hash[@assessment.id]
+      _, end_at, = @personal_timeline_hash[[@assessment.id, submitter_course_user.id]] ||
+                   @reference_times_hash[@assessment.id]
```

Comment:
Does this account for Multiple Reference Timelines (as in the timeline designer)?

Reply:
yes, this functionality considers the personalised timeline to be the priority, then only using reference timeline if there's no personalised one